src/server/db_slice.cc

```diff
@@ -737,6 +737,26 @@ pair<PrimeIterator, ExpireIterator> DbSlice::ExpireIfNeeded(const Context& cntx,
   return make_pair(PrimeIterator{}, ExpireIterator{});
 }
 
+void DbSlice::ExpireAllIfNeeded() {
+  for (unsigned db_index = 0; db_index < db_arr_.size(); db_index++) {
+    if (!db_arr_[db_index])
+      continue;
+    auto& db = *db_arr_[db_index];
+
+    auto cb = [&](ExpireTable::iterator exp_it) {
+      auto it = db.prime.Find(exp_it->first.ToString());
```

Comment:
why do you need to find using `exp_it->first.ToString()` ? `exp_it->first` does not work?

Reply:
No, `Find` expects a key and calls `DoHash` inside. What else could I use?